core/trino-main/src/test/java/io/trino/operator/aggregation/TestDistinctAccumulatorFactory.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.aggregation;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.trino.ExceededMemoryLimitException;
+import io.trino.metadata.TestingFunctionResolution;
+import io.trino.spi.Page;
+import io.trino.sql.tree.QualifiedName;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.OptionalInt;
+
+import static io.trino.ExceededMemoryLimitException.exceededLocalTotalMemoryLimit;
+import static io.trino.block.BlockAssertions.createBooleansBlock;
+import static io.trino.block.BlockAssertions.createLongsBlock;
+import static io.trino.operator.aggregation.AggregationTestUtils.createGroupByIdBlock;
+import static io.trino.sql.planner.plan.AggregationNode.Step.SINGLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.analyzer.TypeSignatureProvider.fromTypes;
+
+public class TestDistinctAccumulatorFactory
+{
+    private static final TestingFunctionResolution testingFunctionResolution = new TestingFunctionResolution();
+    private static final TestingAggregationFunction function = testingFunctionResolution.getAggregateFunction(QualifiedName.of("sum"), fromTypes(BIGINT));
+    private static final AggregatorFactory distinctFactory = function.createDistinctAggregatorFactory(SINGLE, ImmutableList.of(0), OptionalInt.of(1));
+    private static final Long[] longBlock = new Long[20000];
+
+    @BeforeClass
+    public void setUp()
+    {
+        for (int i = 0; i < longBlock.length; i++) {
+            longBlock[i] = (long)i;
+        }
+    }
+
+    @Test(expectedExceptions = ExceededMemoryLimitException.class)
+    public void testUpdateMemory()
+    {
+        Page page = new Page(createLongsBlock(longBlock), createBooleansBlock(true, true, true, true));
```

Comment:
page should have blocks with equal position count, here you have one block with 20k positions and the other with 4

Reply:
Done